cpp/src/quantiles/quantiles.cu

```diff
@@ -157,55 +178,81 @@ namespace{ //unknown
                                                            //(2) for possible types bigger than double, in the future;
                          gdf_context*        ctxt)         //context info
     {
-      return trampoline_exact<T>(col_in, prec, q, t_erased_res, ctxt);
+      return trampoline_exact<T, double>
+                 (col_in, prec, q, t_erased_res, ctxt);
     }
   };
 
   struct trampoline_approx_functor{
+    template <typename UnderLyingT>
+    gdf_error call(gdf_column*  col_in,
+                    double       q,
+                    void*        t_erased_res,
+                    gdf_context* ctxt,
+                    UnderLyingT* data)
+    {
+      return trampoline_exact<UnderLyingT, UnderLyingT>
+                 (col_in, GDF_QUANT_LOWER, q, t_erased_res, ctxt);
+    }
+
     template <typename T>
-    void operator()(gdf_column*  col_in,       //input column;
+    gdf_error operator()(gdf_column*  col_in,       //input column;
                     double       q,            //requested quantile in [0,1]
                     void*        t_erased_res, //type-erased result of same type as column;
                     gdf_context* ctxt)         //context info)
-    {
-      trampoline_approx<T>(col_in, q, t_erased_res, ctxt);
+    { 
+      // ToDo: remove calling "call" function with UnderLying type and directly call "trampoline_exact"

```

Comment:
I think you can remove the `unwrap` if you use SFINAE for `std::is_arithmetic` like is used in `trampoline_exact_functor`.